### DISCUSSION TITLE
Posting CLEF via HTTP "'Newtonsoft.Json.Linq.JValue' is not compatible with expected type 'Newtonsoft.Json.Linq.JObject'

### DISCUSSION THREAD

**Discussion by yooakim** [datalust/seq-tickets#1656] [General]
I am trying to send a message via the [HTTP endpoint](https://docs.datalust.co/docs/posting-raw-events) from Azure Data Factory to Seq. I have read the description of the [CLEF](https://clef-json.org/) format and I am trying to POST this message (the `utcNow()` is part of Azure Data Factory):

    '{"@t":"@{utcNow()}","@mt":"Hello, {User}","User":"nblumhardt"}'

I believe the above should be a valid message , but when I post it Seq rejects it with the following error:

```
[2022-09-26T11:55:05.3035948+00:00 DBG] Rejecting CLEF payload from 40.74.26.153 due to invalid JSON, item could not be parsed; first 81 chars of 81: '{"@t":"2022-09-26T11:55:04.2780189Z","@mt":"Hello, {User}","User":"nblumhardt"}'
Newtonsoft.Json.JsonSerializationException: Deserialized JSON type 'Newtonsoft.Json.Linq.JValue' is not compatible with expected type 'Newtonsoft.Json.Linq.JObject'. Path '', line 1, position 81.
```

Even in the error message it looks as if the payload is ok.  I have two headers in the post request, `X-Seq-ApiKey` and `Content-Type= application/vnd.serilog.clef`

I really appreciate if anybody has any insights into how I can fix this?

**Comment by yooakim** [comment]
It was a user error (as it often is) and the UX in Azure Data Factory isn't very clear... once I change the Azure Data Factory Expression to this:

```
{"@t":"@{utcNow()}","@m":"Hello from ADF, @{pipeline().DataFactory}","@i":"@{pipeline().RunId}", "@DataFactory":"@{pipeline().DataFactory}"}
```

so the issue was with how Azure Data Factory generates JSON from expressions...

**Comment by liammclennan** [comment]
Hi @yooakim ,

I'm glad you found a solution. 

For anyone else who comes across this in the future it is often helpful to test the endpoint directly with something like cURL. 

```
 curl -v -X POST https://<your domain here>/api/events/raw\?clef -H 'Content-Type: application/json' -d '{"@t":"2022-09-26T11:55:04.2780189Z","@mt":"Hello, {User}","User":"nblumhardt"}'
 ```

(I had to escape the `?` in `?clef` because of my shell)
 
 Once you have the message working then you can move on to making sure that this is the message that is being sent.

**Reply by yooakim** [reply]
Yes, good tip. And another one is to look in the ingestion logs of the Seq server. That is what helped my realize the issue with the quoting.

**Reply by liammclennan** [reply]
Thanks @yooakim. That is even better!
